apps/condo/domains/billing/schema/AllResidentBillingReceiptsService.js

```diff
@@ -184,26 +184,36 @@ const AllResidentBillingReceiptsService = new GQLCustomSchema('AllResidentBillin
                 //
                 // Set receipt.paid field and calculate fees
                 //
-                const receiptsWithPayments = []
-                for (const receipt of processedReceipts) {
+                const uniqueContextIds = [...new Set(
+                    processedReceipts
+                        .map(receipt => get(receipt, ['serviceConsumer', 'acquiringIntegrationContext'], null))
+                        .filter(Boolean)
+                )]
+                const formulas = await Promise.all(uniqueContextIds.map(contextId =>
+                    getAcquiringIntegrationContextFormula(context, contextId)
+                ))
+                const formulaMap = {}
+
+                uniqueContextIds.forEach((contextId, index) => {
+                    formulaMap[contextId] = formulas[index]
+                })
+
+                return Promise.all(processedReceipts.map(async receipt => {
```

Comment:
To explicitly point out that we return resolved promises
```suggestion
                return await Promise.all(processedReceipts.map(async receipt => {
```